todo_app/templates/index.html

```diff
@@ -2,17 +2,51 @@
 {% block title %}To-Do App{% endblock %}
 
 {% block content %}
-  <div class="jumbotron">
-    <h1 class="display-4">To-Do App</h1>
-    <p class="lead">Just another to-do app.</p>
+<div class="jumbotron">
+  <h1 class="display-4">To-Do App</h1>
+  <p class="lead">Just another to-do app.</p>
+</div>
+<div class="row">
+  <div class="container mb-3">
+    <form action="/todos" method="post">
+      <div class="mb-3">
+        <label for="title" class="form-label">New Todo</label>
+        <input type="text" id="title" name="title" class="form-control" />
+      </div>
+      <button type="submit" class="btn btn-primary">Add</button>
+    </form>
   </div>
-
-  <div class="row">
-    <div class="col-md-12">
-      <h2>Items</h2>
-      <ul class="list-group mb-4">
-        <!-- List the todo items here -->
-      </ul>
-    </div>
+</div>
+<div class="row">
+  <div class="col-md-12">
+    <h2>Items</h2>
+    <ul class="list-group mb-4">
+      {% for item in items %}
+        <li class="list-group-item align-items-center d-flex justify-content-between">
+          <div class="justify-content-start">
+            {{item.title}}
+          </div>
+          <div class="d-flex">
+            <form action="todos/{{item.id}}" method="post">
+              <div class="form-check mr-4 d-flex align-items-center">
+                <input class="form-check-input" type="checkbox" name="status" value="Complete" onChange="this.form.submit()" {% if item.status=='Complete' %} checked {% endif %}>
```

Comment:
This UI isn't the most intuitive; having the checkbox both represent current state and be a self-submitting form to update the state

Reply:
How would you recommend doing this otherwise? Only alternative I can think of is having the checkbox to mark the todo state as completed and a separate save button to then submit all the changed todos but then you need to maintain state on the client side which seems more complex